fdbclient/NativeAPI.actor.cpp

```diff
@@ -850,14 +853,21 @@ ACTOR static Future<Void> handleTssMismatches(DatabaseContext* cx) {
 				try {
 					tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
 					tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
-
 					if (quarantine) {
-						tr->set(tssQuarantineKeyFor(tssID), LiteralStringRef(""));
+						tr->set(tssQuarantineKeyFor(data.first), LiteralStringRef(""));
 					} else {
-						tr->clear(serverTagKeyFor(tssID));
+						tr->clear(serverTagKeyFor(data.first));
 					}
 					tssMapDB.erase(tr, tssPairID);
 
+					for (const DetailedTSSMismatch& d : data.second) {
+						// <tssid, time, mismatchid> -> mismatch data
+						tssMismatchDB.set(
```

Comment:
If this is the only usage of `tssMismatchDB`, the key template type could be a struct with 3 typed and named members and just have `pack()` and `unpack()` methods defined to convert to/from `Tuple`.  This either won't change the binary representation, or it will improve it, I can't recall what `Codec<>` does for `Tuple` currently, it might actually store it as a packed tuple string inside of another.

Reply:
I just checked `Codec<Tuple>` and it does the good/compatible thing of just using/returning the `Tuple` directly, so changing the key type here to a struct would not change the binary representation, so there's no reason to.

Reply:
Yep I went through the same thought process hahah.